open-sphere-base/mantle/src/main/java/io/opensphere/mantle/icon/impl/gui/IconChooserPanel.java

```diff
@@ -606,6 +614,24 @@ private void displayIconRecords(final List<IconRecord> recList, boolean canCance
         myLoader = new Thread(new BuildIconGridWorker(recList, canCancel));
         myLoader.start();
     }
+    
+    /**
+     * Shows resizes icons and displays loading screen until done.
+     */
+	public void IconResizer(List<IconRecord> recList,int tileWidth,boolean canCancel) 
+	{
+		myTileWidth = tileWidth; 
+        myLoader = new Thread(new BuildIconGridWorker(recList, canCancel));
+        myLoader.start();
+        EventQueueUtilities.runOnEDT(() ->

```

Comment:
I think you might be able to extract the **EventQueueUtilities.runOnEDIT()** to a seperate method since both displayIconRecords and IconResizer use the exact same code